### PR TITLE
Add index page to Pigeonhole, showing shift breakdowns, as well as historical data

### PR DESCRIPTION
**Pull request by SarahKowalik**

### CODE REVIEW

File: lib/influx.rb [hunk 1]
```diff
@@ -234,6 +234,94 @@ def threshold_recommendations(opts)
       end
     end
 
+    def unaddressed_alerts
+      start_date = Date.today.prev_day.strftime('%F')
+      end_date = Date.today.strftime('%F')
+      incidents = find_incidents(start_date, end_date) ##TODO eep
```
Comment (Hobbsee) [inline lib/influx.rb]: @glasnt why is this TODO here?

File: lib/influx.rb [hunk 2]
```diff
@@ -234,6 +234,94 @@ def threshold_recommendations(opts)
...
+        next if incident['incident_key'].nil?
+        entity, check = incident['incident_key'].split(':', 2)
+        member = {
+          'alert_time' => incident["time"],
+          'id' => incident['id'],
+          'entity' => entity,
+          'check'  => check,
+          'ack_by' => incident['acknowledge_by'],
+          'time_to_ack' => incident['time_to_ack'],
+          'time_to_resolve' => incident['time_to_resolve']
```
Comment (Hobbsee) [inline lib/influx.rb]: time_to_resolve makes no sense here - none of these alerts are resolved.

File: lib/influx.rb [hunk 3]
```diff
@@ -234,6 +234,90 @@ def threshold_recommendations(opts)
...
+
+      # We want to show data for one full shift each, plus the current shift.
+      # Due to timezones, this sometimes means we need to look back over the last 3 days
+      shift_matrix = []
+      [
+        Date.today.prev_day.prev_day.strftime('%F'),
+        Date.today.prev_day.strftime('%F'),
+        Date.today.strftime('%F')
+      ].each do |day|
+        shift_times = shifts.each do |_, shift|
```
Comment (nerdyness) [inline lib/influx.rb]: `s/shift_times =//`

File: lib/influx.rb [hunk 4]
```diff
@@ -234,6 +234,90 @@ def threshold_recommendations(opts)
...
+      stat_matrix.flatten!
+
+      aggregate_select_str = "select count(incident_key), " + %w(ack resolve).map { |type|
+        "MEAN(time_to_#{type}) as #{type}_mean, STDDEV(time_to_#{type}) as #{type}_stddev, PERCENTILE(time_to_#{type}, 95) as #{type}_95_percentile"
+      }.join(', ')
+
+      stat_matrix.each do |shift|
+        aggregated = find_incidents(shift[:start], shift[:end],
+                                    :query_select => aggregate_select_str).first || {}
+        ack_percent_in_60s = ack_percent_before_timeout(:start_date => shift[:start], :end_date => shift[:end],
```
Comment (nerdyness) [inline lib/influx.rb]: If the line break here was before `:end_date` the line would be ~86 chars - that seems more reaonable

**Comment by nerdyness** [comment]
Looks good - much easier to digest now :+1:

**Comment by glasnt** [comment]
Updated to address new comments.

**Comment by SarahKowalik** [comment]
Excellent, thanks for the reviews and fixes.

**Pull request merged**